### PR TITLE
Fix check for crypto and OpenSSL

### PR DESCRIPTION
**Pull request by JonnyWhatshisface**
Implements fix for https://github.com/BitchX/BitchX1.3/issues/3 .

Tested and confirmed on MacOS 10.15.7, FreeBSD 13.0-CURRENT and Debian 11.

### CODE REVIEW

**Comment by mkeen** [comment]
FWIW applying this patch in latest debian with openssl 3.0.7 didn't help

**Comment by JonnyWhatshisface** [comment]
> FWIW applying this patch in latest debian with openssl 3.0.7 didn't help

Please give me the version shown in _/etc/debian_version_ and the output of _openssl version_ .

**Comment by antdude** [comment]
Here's mine since I have the same issue in my updated 64-bit Debian (stable; bullseye) PC:

$ cat /etc/debian_version
11.5

$ dpkg -l |grep openssl
ii  openssl                              1.1.1n-0+deb11u3                 amd64        Secure Sockets Layer toolkit - cryptographic utility

**Comment by mkeen** [comment]
```
$ cat /etc/debian_version
bookworm/sid
```

```
$ dpkg -l |grep openssl
ii  libsrt1.5-openssl:amd64                  1.5.1-1                           amd64        Secure Reliable Transport UDP streaming library (OpenSSL flavour)
ii  openssl                                  3.0.7-1                           amd64        Secure Sockets Layer toolkit - cryptographic utility
ii  perl-openssl-defaults:amd64              7+b1                              amd64        version compatibility baseline for Perl OpenSSL packages
ii  python3-openssl                          21.0.0-1                          all          Python 3 wrapper around the OpenSSL library
```

**Comment by antdude** [comment]
Oops. I forgot to show:
$ openssl version
OpenSSL 1.1.1n  15 Mar 2022

**Comment by JonnyWhatshisface** [comment]
I am guessing your compile works fine but you don't have the -s option usable?
 
Looks to be that the issue now is the code is expecting HAVE_SSL be defined, but HAVE_LIBSSL is what's being defined. 

This needs some reworking. 

You can manually hack your way around it for the time being by editing include/defs.h and changing #define HAVE_LIBSSL 1 to #define HAVE_SSL 1 or simply adding the HAVE_SSL define.

**Comment by JonnyWhatshisface** [comment]
Try with the new patch. This is working for me.

**Comment by antdude** [comment]
Newbie question: How do I use this patch?

**Comment by JonnyWhatshisface** [comment]
@mkeen also just for grins: you can use the cp437 emulator to get proper display of all the characters. I'm working on integrating cp437 emulation directly in. My concern is I don't know if @cdedwards is even maintaining this anymore. If not, then I'll simply keep updating my fork...

**Comment by antdude** [comment]
For kicks, I git cloned https://github.com/JonnyWhatshisface/BitchX1.3 into my Debian PC but autogen.sh doesn't see my OpenSSL:

$ time ./autogen.sh
processing .
Running autoheader ...
configure.in:171: warning: AC_LANG_CONFTEST: no AC_LANG_SOURCE call detected in body
../../lib/autoconf/lang.m4:193: AC_LANG_CONFTEST is expanded from...
../../lib/autoconf/general.m4:2672: _AC_LINK_IFELSE is expanded from...
../../lib/autoconf/general.m4:2689: AC_LINK_IFELSE is expanded from...
configure.in:171: the top level
Running autoconf ...
configure.in:171: warning: AC_LANG_CONFTEST: no AC_LANG_SOURCE call detected in body
../../lib/autoconf/lang.m4:193: AC_LANG_CONFTEST is expanded from...
../../lib/autoconf/general.m4:2672: _AC_LINK_IFELSE is expanded from...
../../lib/autoconf/general.m4:2689: AC_LINK_IFELSE is expanded from...
configure.in:171: the top level
Running ./configure ...
Welcome to the BitchX-1.3-git configuration

checking for gcc... gcc
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables...
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether we are using the GNU C compiler... yes
checking whether gcc accepts -g... yes
checking for gcc option to accept ISO C89... none needed
checking for a BSD-compatible install... /usr/bin/install -c
checking whether build environment is sane... yes
checking whether ln -s works... yes
checking for gmake... gmake
checking whether gmake sets $(MAKE)... yes
checking how to run the C preprocessor... gcc -E
checking for grep that handles long lines and -e... /usr/bin/grep
checking for egrep... /usr/bin/grep -E
checking whether gcc needs -traditional... no
checking for library containing strerror... none required
checking for ANSI C header files... yes
checking for sys/types.h... yes
checking for sys/stat.h... yes
checking for stdlib.h... yes
checking for string.h... yes
checking for memory.h... yes
checking for strings.h... yes
checking for inttypes.h... yes
checking for stdint.h... yes
checking for unistd.h... yes
checking minix/config.h usability... no
checking minix/config.h presence... no
checking for minix/config.h... no
checking whether it is safe to define __EXTENSIONS__... yes
checking for gawk... gawk
checking whether compiler understands -fno-strict-aliasing... yes
checking for getpwnam in -lsun... no
checking for inet_addr in -ldgc... no
checking for crypt in -lcrypt... yes
checking for pow in -lm... yes
checking for dirent.h that defines DIR... yes
checking for library containing opendir... none required
checking sys/socket.h usability... yes
checking sys/socket.h presence... yes
checking for sys/socket.h... yes
checking sys/time.h usability... yes
checking sys/time.h presence... yes
checking for sys/time.h... yes
checking sys/fcntl.h usability... yes
checking sys/fcntl.h presence... yes
checking for sys/fcntl.h... yes
checking fcntl.h usability... yes
checking fcntl.h presence... yes
checking for fcntl.h... yes
checking sys/file.h usability... yes
checking sys/file.h presence... yes
checking for sys/file.h... yes
checking netinet/in.h usability... yes
checking netinet/in.h presence... yes
checking for netinet/in.h... yes
checking netdb.h usability... yes
checking netdb.h presence... yes
checking for netdb.h... yes
checking sys/un.h usability... yes
checking sys/un.h presence... yes
checking for sys/un.h... yes
checking sys/filio.h usability... no
checking sys/filio.h presence... no
checking for sys/filio.h... no
checking regex.h usability... yes
checking regex.h presence... yes
checking for regex.h... yes
checking arpa/nameser.h usability... yes
checking arpa/nameser.h presence... yes
checking for arpa/nameser.h... yes
checking dirent.h usability... yes
checking dirent.h presence... yes
checking for dirent.h... yes
checking sys/ndir.h usability... no
checking sys/ndir.h presence... no
checking for sys/ndir.h... no
checking sys/dir.h usability... yes
checking sys/dir.h presence... yes
checking for sys/dir.h... yes
checking ndir.h usability... no
checking ndir.h presence... no
checking for ndir.h... no
checking for net/if.h... yes
checking for arpa/inet.h... yes
checking for resolv.h... yes
checking for stpcpy in string.h... yes
checking for getpgid in unistd.h... no
checking for killpg in signal.h... yes
checking for getpass in unistd.h... yes
checking for errno in errno.h... no
checking for struct linger in sys/socket.h... yes
checking for sun_len in sys/un.h... no
checking for bcopy in string.h... yes
checking whether strsignal is declared... yes
checking whether sys_siglist is declared... yes
checking whether _sys_siglist is declared... yes
checking for gcc option to accept ISO C99... none needed
checking for gcc option to accept ISO Standard C... (cached) none needed
checking for an ANSI C-conforming const... yes
checking for uid_t in sys/types.h... yes
checking for inline... inline
checking for mode_t... yes
checking for off_t... yes
checking for pid_t... yes
checking for size_t... yes
checking return type of signal handlers... void
checking whether time.h and sys/time.h may both be included... yes
checking size of unsigned int... 4
checking library required for resolver functions... -lresolv
checking for the iphlpapi library... no
checking for working alloca.h... yes
checking for alloca... yes
checking whether getpgrp requires zero arguments... yes
checking whether setpgrp takes no argument... yes
checking for strftime... yes
checking for socket... yes
checking for gethostname... yes
checking for gettimeofday... yes
checking for strtoul... yes
checking for strlcpy... no
checking for strlcat... no
checking for stpcpy... yes
checking for vsnprintf... yes
checking for snprintf... yes
checking for setsid... yes
checking for strerror... yes
checking for uname... yes
checking for getrusage... yes
checking for sysconf... yes
checking for getpgid... yes
checking for killpg... yes
checking for getlogin... yes
checking for realpath... yes
checking for fchdir... yes
checking for getpass... yes
checking for fpathconf... yes
checking for getpwent... yes
checking for setvbuf... yes
checking for select... yes
checking for mkstemp... yes
checking for memmove... yes
checking for scandir... yes
checking for gethostbyname... yes
checking for connect... yes
checking for inet_aton... yes
checking for non-blocking type... posix
checking for a list of signal names... sys_siglist
checking whether to enable GTK support... no
checking whether to enable Win32 GUI support... no
checking whether to enable OS/2 PM support... no
checking whether to enable sound support... no
checking whether to enable SSL support... if available
checking for ERR_get_error in -lcrypto... no
configure: WARNING: OpenSSL not found, will not have SSL support.
checking whether to use use tgetent or setupterm... setupterm
checking for setupterm in -lncurses... yes
checking for tparm in ncurses.h... yes
checking for tparm in -lncurses... yes
checking ncurses.h usability... yes
checking ncurses.h presence... yes
checking for ncurses.h... yes
checking ncurses/termcap.h usability... no
checking ncurses/termcap.h presence... no
checking for ncurses/termcap.h... no
checking termcap.h usability... yes
checking termcap.h presence... yes
checking for termcap.h... yes
checking for tputs in ncurses/termcap.h... no
checking for tputs in termcap.h... (cached) no
checking for tputs in -lncurses... yes
checking whether to enable Tcl support... no
checking for unix mail directory... /var/spool/mail
checking for default server list... none
checking whether to enable IPv6 support... no
checking whether to enable SOCKS support... no
checking whether to enable CD-ROM support... no
checking whether to enable dmalloc debugging support... no
checking whether to enable plugin support... not building with plugin support - you will not be able to use plugins
configure: creating ./config.status
config.status: creating Makefile
config.status: WARNING:  'Makefile.in' seems to ignore the --datarootdir setting
config.status: creating bx-conf/Makefile
config.status: WARNING:  'bx-conf/Makefile.in' seems to ignore the --datarootdir setting
config.status: creating doc/BitchX.bat
config.status: creating doc/Makefile
config.status: creating dll/Makefile
config.status: WARNING:  'dll/Makefile.in' seems to ignore the --datarootdir setting
config.status: creating dll/abot/Makefile
config.status: WARNING:  'dll/abot/Makefile.in' seems to ignore the --datarootdir setting
config.status: creating dll/acro/Makefile
config.status: WARNING:  'dll/acro/Makefile.in' seems to ignore the --datarootdir setting
config.status: creating dll/aim/Makefile
config.status: WARNING:  'dll/aim/Makefile.in' seems to ignore the --datarootdir setting
config.status: creating dll/aim/toc/Makefile
config.status: WARNING:  'dll/aim/toc/Makefile.in' seems to ignore the --datarootdir setting
config.status: creating dll/amp/Makefile
config.status: WARNING:  'dll/amp/Makefile.in' seems to ignore the --datarootdir setting
config.status: creating dll/arcfour/Makefile
config.status: WARNING:  'dll/arcfour/Makefile.in' seems to ignore the --datarootdir setting
config.status: creating dll/autocycle/Makefile
config.status: WARNING:  'dll/autocycle/Makefile.in' seems to ignore the --datarootdir setting
config.status: creating dll/blowfish/Makefile
config.status: WARNING:  'dll/blowfish/Makefile.in' seems to ignore the --datarootdir setting
config.status: creating dll/cavlink/Makefile
config.status: WARNING:  'dll/cavlink/Makefile.in' seems to ignore the --datarootdir setting
config.status: creating dll/cdrom/Makefile
config.status: WARNING:  'dll/cdrom/Makefile.in' seems to ignore the --datarootdir setting
config.status: creating dll/encrypt/Makefile
config.status: WARNING:  'dll/encrypt/Makefile.in' seems to ignore the --datarootdir setting
config.status: creating dll/europa/Makefile
config.status: WARNING:  'dll/europa/Makefile.in' seems to ignore the --datarootdir setting
config.status: creating dll/fserv/Makefile
config.status: WARNING:  'dll/fserv/Makefile.in' seems to ignore the --datarootdir setting
config.status: creating dll/hint/Makefile
config.status: WARNING:  'dll/hint/Makefile.in' seems to ignore the --datarootdir setting
config.status: creating dll/identd/Makefile
config.status: WARNING:  'dll/identd/Makefile.in' seems to ignore the --datarootdir setting
config.status: creating dll/nap/Makefile
config.status: WARNING:  'dll/nap/Makefile.in' seems to ignore the --datarootdir setting
config.status: creating dll/nicklist/Makefile
config.status: WARNING:  'dll/nicklist/Makefile.in' seems to ignore the --datarootdir setting
config.status: creating dll/pkga/Makefile
config.status: WARNING:  'dll/pkga/Makefile.in' seems to ignore the --datarootdir setting
config.status: creating dll/possum/Makefile
config.status: WARNING:  'dll/possum/Makefile.in' seems to ignore the --datarootdir setting
config.status: creating dll/qbx/Makefile
config.status: WARNING:  'dll/qbx/Makefile.in' seems to ignore the --datarootdir setting
config.status: creating dll/qmail/Makefile
config.status: WARNING:  'dll/qmail/Makefile.in' seems to ignore the --datarootdir setting
config.status: creating dll/wavplay/Makefile
config.status: WARNING:  'dll/wavplay/Makefile.in' seems to ignore the --datarootdir setting
config.status: creating dll/xmms/Makefile
config.status: WARNING:  'dll/xmms/Makefile.in' seems to ignore the --datarootdir setting
config.status: creating source/Makefile
config.status: WARNING:  'source/Makefile.in' seems to ignore the --datarootdir setting
config.status: creating include/defs.h
config.status: executing default commands

BitchX (c) 1996-2012 Colten Edwards et al
----------------------------------------------------------

The configuration script has finished. You should look through
"include/config.h" and make any changes you would like to make.

Now type "gmake" to compile BitchX.

real    0m12.543s
user    0m8.565s
sys     0m3.558s

Am I missing something or doing something wrong? :(

**Comment by JonnyWhatshisface** [comment]
It's because:

`checking for ERR_get_error in -lcrypto... no`

Do you have /usr/include/openssl/err.h present? Do you have libssl-dev installed? Check with:

`dpkg -l libssl-dev`

**Comment by antdude** [comment]
Ah hah. I don't have that package installed. Now, it works!  Ooh, I should try the older versions to see if they can compile now.

**Comment by antdude** [comment]
Darn it! git clone https://github.com/BitchX/BitchX1.3 failed to compile with OpenSSL packages:
...
checking whether to enable SSL support... if available
checking for SSLeay in -lcrypto... no
configure: WARNING: OpenSSL not found, will not have SSL support.
...

I guess I will use JonnyWhatshisface's git for now. When do we expect his git changes patch to merge into the official BitchX's git?

**Review by cdedwards — APPROVED** [review]

File: configure.in
Comment (Stuff4ik) [inline configure.in:1]: 1

Comment (Stuff4ik) [reply configure.in:1]: .

**Pull request merged**